### PR TITLE
Add restriction on saving django_settings_override - EDLY_2403

### PR DESCRIPTION
**Pull request by taimoor-ahmed-1**
**Description:** Adds restrictions in site configuration to make sure only allowed settings in `DJANGO_SETTINGS_OVERRIDE` are being overriden. 
**JIRA:**
https://edlyio.atlassian.net/browse/EDLY-2403

**Visible Changes:**
Screenshot 2021-01-07 at 2 57 04 PM

### CODE REVIEW

File: ecommerce/extensions/edly_ecommerce_app/helpers.py [hunk 1]
```diff
@@ -109,3 +111,37 @@ def user_is_course_creator(request):
 
     decoded_cookie_data = decode_edly_user_info_cookie(edly_user_info_cookie)
     return decoded_cookie_data.get('is_course_creator', False)
+
+def clean_django_settings_override(django_settings_override):
```
Comment (iam-mhaseeb) [inline ecommerce/extensions/edly_ecommerce_app/helpers.py:115]: Can we rename it `validate_django_settings_override`?

**Review by iam-mhaseeb — APPROVED** [review]

**Review by zubair-arbi — APPROVED** [review]
Change looks good to me. 👍 
@rehan99000 Please have a look

File: ecommerce/extensions/edly_ecommerce_app/tests/test_helpers.py [hunk 2]
```diff
@@ -1,17 +1,20 @@
 from django.conf import settings
 from django.test import RequestFactory
+from django.core.exceptions import ValidationError
```
Comment (rehan99000) [inline ecommerce/extensions/edly_ecommerce_app/tests/test_helpers.py:3]: reorder imports in alphabetical order

**Review by rehan99000 — APPROVED** [review]
just one minor change.

File: ecommerce/extensions/edly_ecommerce_app/tests/test_helpers.py (hunk 2 above)
Comment (taimoor-ahmed-1) [reply ecommerce/extensions/edly_ecommerce_app/tests/test_helpers.py:2]: Done.

File: ecommerce/core/models.py [hunk 3]
```diff
@@ -23,6 +23,7 @@
 from ecommerce.core.utils import log_message_and_raise_validation_error
 from ecommerce.extensions.payment.exceptions import ProcessorNotFoundError
 from ecommerce.extensions.payment.helpers import get_processor_class, get_processor_class_by_name
+from ecommerce.extensions.edly_ecommerce_app.helpers import clean_django_settings_override
```
Comment (burhandodhy) [inline ecommerce/core/models.py:26]: Move this line before `from ecommerce.extensions.payment.exceptions import ProcessorNotFoundError`

**Review by burhandodhy — APPROVED** [review]

File: ecommerce/core/models.py (hunk 3 above)
Comment (taimoor-ahmed-1) [reply ecommerce/core/models.py]: Done.

**Pull request merged**